src/Orchard/Data/Migration/AutomaticDataMigrations.cs

```diff
@@ -25,30 +29,34 @@ IFeatureManager featureManager
         public ILogger Logger { get; set; } 
 
         public void Activated() {
-
-            // Let's make sure that the basic set of features is enabled.  If there are any that are not enabled, then let's enable them first.
-            var theseFeaturesShouldAlwaysBeActive = new[] {
-                "Common", "Containers", "Contents", "Dashboard", "Feeds", "Navigation", "Scheduling", "Settings", "Shapes", "Title"
-            };
-
-            var enabledFeatures = _featureManager.GetEnabledFeatures().Select(f => f.Id).ToList();
-            var featuresToEnable = theseFeaturesShouldAlwaysBeActive.Where(shouldBeActive => !enabledFeatures.Contains(shouldBeActive)).ToList();
-            if (featuresToEnable.Any()) {
-                _featureManager.EnableFeatures(featuresToEnable, true);
-            }
-
-            foreach (var feature in _dataMigrationManager.GetFeaturesThatNeedUpdate()) {
-                try {
-                    _dataMigrationManager.Update(feature);
+            using(var @lock = _distributedLockService.AcquireLock(GetType().FullName, TimeSpan.FromMinutes(30), TimeSpan.FromMilliseconds(250))) {
+                if (@lock == null)
+                    return;

```

Comment:
Looking at your comment below, why do you think it is better to throw an exception than returning null? Is that a pattern? And if we did change the method to throw, would you surround the `using` statement with a try/catch block, or not use the `using` statement at all?